service-worker.js

```diff
@@ -0,0 +1,254 @@
+          resolve(JSON.parse(result.authorization));
+        } else {
+          resolve({});
+        }
+      });
+    });
+  }
+
+  /**
+   * Starts the authorization flow and retrieves an access token upon success.
+   */
+  function getAccessToken() {
+    return new Promise((resolve, reject) => {
+      var url;
+      const pkce = getPkce();
+      const state = getState();
+
+      url =
+        `${authorizeUrl}?client_id=${clientId}&response_type=code&code_challenge_method=S256` +
+        `&code_challenge=${
+          pkce.challenge
+        }&redirect_uri=${chrome.identity.getRedirectURL(
+          "oauth2"
+        )}&state=${state}`;
+
+      // Starts an authorization flow at the specified URL.
+      // - https://developer.chrome.com/apps/identity#method-launchWebAuthFlow
+      chrome.identity.launchWebAuthFlow(
+        {
+          url: url,
+          interactive: true,
+        },
+
+        // Retrieves the value of the `code` and `state`URL parameter.
+        function handleRedirect(authCodeUrl) {
+          // Handle any errors encountered.
+          if (chrome.runtime.lastError) {
+            console.log(
+              chrome.runtime.lastError.message +
+                " Is your SKY API Application redirect URI set to " +
+                chrome.identity.getRedirectURL("oauth2") +
+                "?"
+            );
+            return reject({
+              error:
+                chrome.runtime.lastError.message +
+                " Check the Background Page console for more info.",
+            });
+          }
+
+          const authCodeFragments = getUrlParams(
+            "?" + authCodeUrl.split("?")[1]
+          );
+          const code = authCodeFragments.code;
+          const responseState = authCodeFragments.state;
+
+          if (responseState !== state) {
+            reject("State parameter was not valid");
+          }
+
+          if (!code) {
+            reject("Error getting authorization code");
+          }
+
+          fetch(tokenUrl, {
+            method: "post",
+            body: new URLSearchParams({
+              client_id: clientId,
+              grant_type: "authorization_code",
+              redirect_uri: chrome.identity.getRedirectURL("oauth2"),
+              code: code,
+              code_verifier: pkce.verifier,
+            }),
+          })
+            .then((response) => response.json())
+            .then((body) => {
+              if (!body.access_token) {
+                reject(body);
+              } else {
+                chrome.storage.local.set({
+                  authorization: JSON.stringify(body),
+                });
+                resolve(body);
+              }
+            })
+            .catch(reject);
+        }
+      );
+    });
+  }
+
+  /**
+   * Makes a request to SKY API Constituent Search endpoint:
+   *  - https://developer.sky.blackbaud.com/docs/services/56b76470069a0509c8f1c5b3/operations/56b76471069a050520297727
```

Comment:
https://developer.sky.blackbaud.com/docs/services/56b76470069a0509c8f1c5b3/operations/SearchConstituent

Reply:
The link to the endpoint reference should be updated now